src/api/0x/0x.spec.ts

```diff
@@ -89,5 +88,37 @@ test('Invalid: Get Price Quote', async () => {
     { status: 400 }
   )
 
+  // expect(fetchMock).toHaveBeenCalledTimes(1)
+  await expect(cowSdk.zeroXApi.getQuote(query)).rejects.toThrowError(ZeroXError)
```

Comment:
Sorry to be that guy, but, you I am that guy ;)

Anyway, it looks much cleaner, nice!

But you have removed couple of steps from the test:

```
    expect(error.message).toEqual(
      'Price fetch failed. This may be due to a server or network connectivity issue. Please try again later.'
    )
    expect(error.error_code).toEqual('400')
    expect(fetchMock).toHaveBeenCalledTimes(1)
```

It might or might not be possible to do multiple checks in a single `toThrowError` statement, but you can use another statement to check the mock has been called

Reply:
the rejects approach is a bit limiting ngl, maybe I'm missing sth but that approach doesn't allow me to make multiple calls - hence why i thought the try/catch was ok.

actually curious as to why you don't like that approach?

Reply:
We talked about it over our last sync and I'm fine with the try catch approach as long as it fails in case the the exception isn't thrown as expected